src/components/product_copier/product_copier.component.yaml

```diff
@@ -0,0 +1,46 @@
+---
+description: |
+  Given two locations and a list of source/destination IDs, fetches Data_Product entries from one location and sends/copies them to another upon receiving a Tick.
+
+  The use case in mind was for the two locations to be databases (e.g. Product_Database instances), and for this component to take snapshots of a source database at a fixed interval. The idea is that if the values stored in source database is constantly in flux, then the destination database could provide a stable view of the source -- within a tick, the values in the destination database will not change between reads.
+execution: passive
+with:
+  - Product_Mapping
+preamble: |
+  type Product_Mapping_Array is array (Natural range <>) of Product_Mapping.T;
+  type Product_Mapping_Array_Access is access all Product_Mapping_Array;
+init:
+  description: |
+    At initialization, this component requires a list of source/destination pairs of data products to copy.
+  parameters:
+    - name: Products_To_Copy
+      type: Product_Mapping_Array_Access
+      description: |
+        The list of mappings to be copied by this component every tick. Raises an error on Init if the list is null, as well as if two mappings share a destination.
+
+        You must pass a reference (using 'Access) to this function, and as a consequence you can't declare your array of mappings within, say, a function's declarative region, since it must not be garbage collected. Either declare it in a package or use a dynamic allocation.
```

Comment:
This isn't actually true. It can be declared within the declarative region as long as a local access type is take from it. There is also no garbage collection in the traditional sense (unless you mean it gets deallocated from the stack).